pkg/backend-storage/backend-storage.go

```diff
@@ -0,0 +1,75 @@
+package backendstorage
+
+import (
+	"context"
+	"fmt"
+
+	virtconfig "kubevirt.io/kubevirt/pkg/virt-config"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	corev1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/kubecli"
+)
+
+const (
+	PVCPrefix = "persistent-state-for-"
+	PVCSize   = "10Mi"
+)
+
+func HasPersistentTPMDevice(vmi *corev1.VirtualMachineInstance) bool {
+	if vmi.Spec.Domain.Devices.TPM != nil &&
+		vmi.Spec.Domain.Devices.TPM.Persistent != nil &&
+		*vmi.Spec.Domain.Devices.TPM.Persistent {
+		return true
+	}
+
+	return false
+}
+
+func isBackendStorageNeeded(vmi *corev1.VirtualMachineInstance) bool {
+	return HasPersistentTPMDevice(vmi)
+}
+
+func CreateIfNeeded(vmi *corev1.VirtualMachineInstance, clusterConfig *virtconfig.ClusterConfig, client kubecli.KubevirtClient) (bool, error) {
+	if !isBackendStorageNeeded(vmi) {
+		return true, nil
+	}
+
+	_, err := client.CoreV1().PersistentVolumeClaims(vmi.Namespace).Get(context.Background(), PVCPrefix+vmi.Name, metav1.GetOptions{})
+	if err == nil {
+		return true, nil
+	}
+	if !errors.IsNotFound(err) {
+		return false, err
+	}
+
+	modeFile := v1.PersistentVolumeFilesystem
+	storageClass := clusterConfig.GetVMStateStorageClass()
+	if storageClass == "" {
+		return false, fmt.Errorf("backend VM storage requires a backend storage class defined in the custom resource")
+	}
+	pvc := &v1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: PVCPrefix + vmi.Name,
+		},
+		Spec: v1.PersistentVolumeClaimSpec{
+			AccessModes: []v1.PersistentVolumeAccessMode{v1.ReadWriteMany},
+			Resources: v1.ResourceRequirements{
+				Requests: v1.ResourceList{v1.ResourceStorage: resource.MustParse(PVCSize)},
+			},
+			StorageClassName: &storageClass,
+			VolumeMode:       &modeFile,
+		},
+	}
+
+	_, err = client.CoreV1().PersistentVolumeClaims(vmi.Namespace).Create(context.Background(), pvc, metav1.CreateOptions{})
```

Comment:
This would fail also if the PVC already exist, do we want this?Or in this case we simply skip it?

Reply:
The PVC shouldn't exist, since we tested for it earlier in the function, and it shouldn't be called multiple times in parallel.
I can still check for `errors.IsAlreadyExists(err)` and succeed in that case.